File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-eq-shared-5-4/ReturnSwapEarlyTermination.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class ReturnSwapEarlyTermination : ISerialized
    {
        public ReturnSwapEarlyTermination(XmlNode xmlNode)
        {
            XmlNode partyReferenceNode = xmlNode.SelectSingleNode("partyReference");
            
            if (partyReferenceNode != null)
            {
                if (partyReferenceNode.Attributes["href"] != null || partyReferenceNode.Attributes["id"] != null) 
                {
                    if (partyReferenceNode.Attributes["id"] != null) 
                    {
                        partyReferenceIDRef_ = partyReferenceNode.Attributes["id"].Value;
                        PartyReference ob = new PartyReference(partyReferenceNode);
                        IDManager.SetID(partyReferenceIDRef_, ob);
                    }
                    else if (partyReferenceNode.Attributes["href"] != null)
                    {
                        partyReferenceIDRef_ = partyReferenceNode.Attributes["href"].Value;
                    }
                    else
                    {
                        partyReference_ = new PartyReference(partyReferenceNode);
                    }
                }
                else
                {
                    partyReference_ = new PartyReference(partyReferenceNode);
                }
            }
            
        
            XmlNode startingDateNode = xmlNode.SelectSingleNode("startingDate");
            
            if (startingDateNode != null)
            {
                if (startingDateNode.Attributes["href"] != null || startingDateNode.Attributes["id"] != null) 
                {
                    if (startingDateNode.Attributes["id"] != null) 
                    {
                        startingDateIDRef_ = startingDateNode.Attributes["id"].Value;
                        StartingDate ob = new StartingDate(startingDateNode);
                        IDManager.SetID(startingDateIDRef_, ob);
                    }
                    else if (startingDateNode.Attributes["href"] != null)
                    {
                        startingDateIDRef_ = startingDateNode.Attributes["href"].Value;
                    }
                    else
                    {
                        startingDate_ = new StartingDate(startingDateNode);
                    }
                }
                else
                {
                    startingDate_ = new StartingDate(startingDateNode);
                }
            }
            
        
        }
        
    
        #region PartyReference_
        private PartyReference partyReference_;
        public PartyReference PartyReference_
        {
            get
            {
                if (this.partyReference_ != null)
                {
                    return this.partyReference_; 
                }
                else if (this.partyReferenceIDRef_ != null)
                {
                    partyReference_ = IDManager.getID(partyReferenceIDRef_) as PartyReference;
                    return this.partyReference_; 
                }
                else
                {
                      return this.partyReference_; 
                }
            }
            set
            {
                if (this.partyReference_ != value)
                {
                    this.partyReference_ = value;
                }
            }
        }
        #endregion
        
        public string partyReferenceIDRef_ { get; set; }
        #region StartingDate_
        private StartingDate startingDate_;
        public StartingDate StartingDate_
        {
            get
            {
                if (this.startingDate_ != null)
                {
                    return this.startingDate_; 
                }
                else if (this.startingDateIDRef_ != null)
                {
                    startingDate_ = IDManager.getID(startingDateIDRef_) as StartingDate;
                    return this.startingDate_; 
                }
                else
                {
                      return this.startingDate_; 
                }
            }
            set
            {
                if (this.startingDate_ != value)
                {
                    this.startingDate_ = value;
                }
            }
        }
        #endregion
        
        public string startingDateIDRef_ { get; set; }
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-eq-shared-5-4/ReturnSwapEarlyTermination.cpp
// ReturnSwapEarlyTermination.cpp 
#include "ReturnSwapEarlyTermination.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

ReturnSwapEarlyTermination::ReturnSwapEarlyTermination(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //partyReferenceNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* partyReferenceNode = xmlNode->FirstChildElement("partyReference");

   if(partyReferenceNode){partyReferenceIsNull_ = false;}
   else{partyReferenceIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- partyReferenceNode , address : " << partyReferenceNode << std::endl;
   #endif
   if(partyReferenceNode)
   {
      if(partyReferenceNode->Attribute("href") || partyReferenceNode->Attribute("id"))
      {
          if(partyReferenceNode->Attribute("id"))
          {
             partyReferenceIDRef_ = partyReferenceNode->Attribute("id");
             partyReference_ = boost::shared_ptr<PartyReference>(new PartyReference(partyReferenceNode));
             partyReference_->setID(partyReferenceIDRef_);
             IDManager::instance().setID(partyReferenceIDRef_,partyReference_);
          }
          else if(partyReferenceNode->Attribute("href")) { partyReferenceIDRef_ = partyReferenceNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { partyReference_ = boost::shared_ptr<PartyReference>(new PartyReference(partyReferenceNode));}
   }

   //startingDateNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* startingDateNode = xmlNode->FirstChildElement("startingDate");

   if(startingDateNode){startingDateIsNull_ = false;}
   else{startingDateIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- startingDateNode , address : " << startingDateNode << std::endl;
   #endif
   if(startingDateNode)
   {
      if(startingDateNode->Attribute("href") || startingDateNode->Attribute("id"))
      {
          if(startingDateNode->Attribute("id"))
          {
             startingDateIDRef_ = startingDateNode->Attribute("id");
             startingDate_ = boost::shared_ptr<StartingDate>(new StartingDate(startingDateNode));
             startingDate_->setID(startingDateIDRef_);
             IDManager::instance().setID(startingDateIDRef_,startingDate_);
          }
          else if(startingDateNode->Attribute("href")) { startingDateIDRef_ = startingDateNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { startingDate_ = boost::shared_ptr<StartingDate>(new StartingDate(startingDateNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<PartyReference> ReturnSwapEarlyTermination::getPartyReference()
{
   if(!this->partyReferenceIsNull_){
        if(partyReferenceIDRef_ != ""){
             return boost::shared_static_cast<PartyReference>(IDManager::instance().getID(partyReferenceIDRef_));
        }else{
             return this->partyReference_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<PartyReference>();
   }
}
boost::shared_ptr<StartingDate> ReturnSwapEarlyTermination::getStartingDate()
{
   if(!this->startingDateIsNull_){
        if(startingDateIDRef_ != ""){
             return boost::shared_static_cast<StartingDate>(IDManager::instance().getID(startingDateIDRef_));
        }else{
             return this->startingDate_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<StartingDate>();
   }
}
}

